standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -6038,17 +6039,26 @@ public void alter_table_with_environment_context(final String dbname,
       throws InvalidOperationException, MetaException {
     String[] parsedDbName = parseDbName(dbname, conf);
     alter_table_core(parsedDbName[CAT_NAME], parsedDbName[DB_NAME],
-        name, newTable, envContext, null, null, null);
+        name, newTable, envContext, null, null, null, null, null);
   }
 
   private void alter_table_core(String catName, String dbname, String name, Table newTable,
-                                EnvironmentContext envContext, String validWriteIdList, List<String> processorCapabilities, String processorId)
+                                EnvironmentContext envContext, String validWriteIdList, List<String> processorCapabilities,
+                                String processorId, String expectedPropertyKey, String expectedPropertyValue)
```

Comment:
Did you consider allowing a list of table properties here?

Reply:
Iceberg does not explicitly need this, and it would somewhat complicate the code. So I voted down this. Still I do not feel too strongly either way, so if you think it could be useful, I am open to add multiple checks based on a map (do we have map in thrift?)

Reply:
After some more thought, I would opt for keeping the PR as it is - checking only for a single property.
If someone would need more properties to check, then they still could create an uber property - as a hash of the relevant properties -, and use the `EXPECTED_PARAMETER_KEY` and `EXPECTED_PARAMETER_VALUE` to check for the changes of this uber property. And if it is that hard for them, they could still add a new feature later.

Reply:
Though I still think its cleaner to support multi-properties transactional check, given Table has properties map, I'm leave it to you and @prasanthj